src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -137,9 +138,26 @@ private static IQueryable<IPackage> execute_package_search(ChocolateyConfigurati
                         .AsQueryable();
             }
 
-            results = configuration.ListCommand.OrderByPopularity ? 
-                 results.OrderByDescending(p => p.DownloadCount).ThenBy(p => p.Id)
-                 : results;
+
+            switch (configuration.ListCommand.OrderBy)
+            {
+                default:
+                case PackageOrder.name:
+                    results = results.OrderBy(p => p.Id);
+                    break;
+
+                case PackageOrder.title:
+                    results = results.OrderBy(p => p.Title).ThenBy(p => p.Id);
+                    break;
+
+                case PackageOrder.popularity:
+                    results = results.OrderByDescending(p => p.DownloadCount).ThenBy(p => p.Id);
+                    break;
+
+                case PackageOrder.lastpublished:
+                    results = results.OrderByDescending(p => p.Published).ThenBy(p => p.Id);
+                    break;
+            }

```

Comment:
This is likely to conflict with recent work done in the code base

Reply:
I rebased onto master, didn't find any conflicts. Maybe you are talking about logic conflicts, are you? If so, do I need to address those or is it easier if you do this?